src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultTagHelperDescriptorBuilder.cs

```diff
@@ -176,6 +177,9 @@ public override TagHelperDescriptor Build()
         var tagMatchingRules = _tagMatchingRuleBuilders.BuildAllOrEmpty(s_tagMatchingRuleSetPool);
         var attributes = _attributeBuilders.BuildAllOrEmpty(s_boundAttributeSetPool);
 
+        _metadata.AddIfMissing(TagHelperMetadata.Runtime.Name, TagHelperConventions.DefaultKind);
```

Comment:
Is adding this in everywhere going to change the math on our special-cased types? IE, should we go from 1-4 now, instead of 1-3, to accomodate this extra kind?

Reply:
Also, should we special case the 1-length case, ie only has the runtime name, and pool the default value always?

Reply:
Great suggestion!

Reply:
FWIW, there aren't any cases in production code where this happens. TagHelperDescriptors tend to get 6 or more bits of metadata.